File: app/src/main/java/me/iacn/biliroaming/hooker/TeenagersModeHook.java
package me.iacn.biliroaming.hooker;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;

import de.robv.android.xposed.XC_MethodHook;
import me.iacn.biliroaming.XposedInit;

import static de.robv.android.xposed.XposedHelpers.findAndHookMethod;
import static me.iacn.biliroaming.Constant.TAG;

/**
 * Created by iAcn on 2019/12/15
 * Email i@iacn.me
 */
public class TeenagersModeHook extends BaseHook {

    public TeenagersModeHook(ClassLoader classLoader) {
        super(classLoader);
    }

    @Override
    public void startHook() {
        if (!XposedInit.sPrefs.getBoolean("teenagers_mode_dialog", false)) return;
        Log.d(TAG, "startHook: TeenagersMode");

        findAndHookMethod("com.bilibili.teenagersmode.ui.TeenagersModeDialogActivity",
                mClassLoader, "onCreate", Bundle.class, new XC_MethodHook() {
                    @Override
                    protected void afterHookedMethod(MethodHookParam param) throws Throwable {
                        Activity activity = (Activity) param.thisObject;
                        activity.finish();
                        Log.d(TAG, "Teenagers mode dialog has been closed");
                    }
                });
    }
}

File: app/src/main/java/me/iacn/biliroaming/hook/TeenagersModeHook.kt
package me.iacn.biliroaming.hook

import android.app.Activity
import android.os.Bundle
import me.iacn.biliroaming.utils.Log
import me.iacn.biliroaming.utils.hookAfterMethod
import me.iacn.biliroaming.utils.sPrefs

/**
 * Created by iAcn on 2019/12/15
 * Email i@iacn.me
 */
class TeenagersModeHook(classLoader: ClassLoader) : BaseHook(classLoader) {
    override fun startHook() {
        if (!sPrefs.getBoolean("teenagers_mode_dialog", false)) return
        Log.d("startHook: TeenagersMode")
        "com.bilibili.teenagersmode.ui.TeenagersModeDialogActivity".hookAfterMethod(
                mClassLoader, "onCreate", Bundle::class.java) { param ->
            val activity = param.thisObject as Activity
            activity.finish()
            Log.d("Teenagers mode dialog has been closed")
            Log.toast("已关闭青少年模式对话框")
        }
    }
}